agent/agentcontainers/api_test.go

```diff
@@ -146,6 +150,136 @@ func (w *fakeWatcher) sendEventWaitNextCalled(ctx context.Context, event fsnotif
 func TestAPI(t *testing.T) {
 	t.Parallel()
 
+	// List tests the API.getContainers method using a mock
+	// implementation. It specifically tests caching behavior.
+	t.Run("List", func(t *testing.T) {
```

Comment:
This test was moved from internal to here. In the process we had to create WithCacheDuration and change the implementation from calling internal `getContainers` to using the API endpoint (which seems reasonable). Otherwise the test remains unchanged.